### PR TITLE
Add String.IndexOf to well known functions

### PR DESCRIPTION
**Pull request by cdmihai**
This one gets hit a lot in a quickbuild repo I've been debugging through and the Expander exception was getting annoying :)

### CODE REVIEW

File: src/Build/Evaluation/Expander.cs [hunk 1]
```diff
@@ -4242,6 +4250,44 @@ private static bool TryGetArg(object[] args, out string arg0)
...
+            private static readonly Dictionary<string, StringComparison> StringComparisonNames =
+                new Dictionary<string, StringComparison>(StringComparer.Ordinal)
+                {
+                    {"System.StringComparison.OrdinalIgnoreCase", StringComparison.OrdinalIgnoreCase},
+                    {"System.StringComparison.Ordinal", StringComparison.Ordinal},
+                    {"System.StringComparison.CurrentCulture", StringComparison.CurrentCulture},
+                    {"System.StringComparison.CurrentCultureIgnoreCase", StringComparison.CurrentCultureIgnoreCase},
+                    {"System.StringComparison.InvariantCulture", StringComparison.InvariantCulture},
+                    {"System.StringComparison.InvariantCultureIgnoreCase", StringComparison.InvariantCultureIgnoreCase},
+                    {"OrdinalIgnoreCase", StringComparison.OrdinalIgnoreCase},
```
Comment (rainersigwald) [inline src/Build/Evaluation/Expander.cs:4262]: Is it legal to do, e.g. `StringComparison.OrdinalIgnoreCase`?

Comment (rainersigwald) [reply src/Build/Evaluation/Expander.cs:4262]: Or see/abstract https://github.com/microsoft/msbuild/pull/5342/files#diff-dee96bf621e4ed93f0db318ca71d4d35R3572

File: src/Build/Evaluation/Expander.cs [hunk 2]
```diff
@@ -4242,6 +4250,44 @@ private static bool TryGetArg(object[] args, out string arg0)
                 return arg0 != null;
             }
 
+            private static readonly Dictionary<string, StringComparison> StringComparisonNames =
```
Comment (ladipro) [inline src/Build/Evaluation/Expander.cs:4253]: A drop in an ocean but still, it's wasteful to create the dictionary on each class init when it's used only for evaluating one specific function. Can you please initialize it lazily? Also, shouldn't it be named something like ```s_stringComparisonNames``` to match the convention in other private static data in this class?

Comment (cdmihai) [reply src/Build/Evaluation/Expander.cs]: It happens once per process, probably not noticeable. But, I replaced the dictionary with another approach that was already being used in the same set of overloads.

File: src/Build/Evaluation/Expander.cs [hunk 3]
```diff
@@ -4242,6 +4240,33 @@ private static bool TryGetArg(object[] args, out string arg0)
...
+                arg0 = args[0] as string;
+
+                if (arg0 == null || !(args[1] is string comparisonType))
+                {
+                    arg1 = default;
+                    return false;
+                }
+
+                // Allow fully-qualified enum, e.g. "System.StringComparison.OrdinalIgnoreCase"
+                if (comparisonType.Contains("."))
```
Comment (ladipro) [inline src/Build/Evaluation/Expander.cs:4262]: super-nit: ```'.'``` instead of ```"."```

Comment (cdmihai) [reply src/Build/Evaluation/Expander.cs:4262]:
```suggestion
                if (comparisonType.Contains('.'))
```

**Pull request merged**